vendor/bat-native-ledger/src/bat/ledger/ledger.cc

```diff
@@ -537,14 +538,16 @@ bool Grant::loadFromJson(const std::string& json) {
     error = !(d.HasMember("altcurrency") && d["altcurrency"].IsString() &&
         d.HasMember("probi") && d["probi"].IsString() &&
         d.HasMember("promotionId") && d["promotionId"].IsString() &&
-        d.HasMember("expiryTime") && d["expiryTime"].IsUint64());
+        d.HasMember("expiryTime") && d["expiryTime"].IsUint64()) &&
+        d.HasMember("type") && d["type"].IsString();
```

Comment:
we need to move this one out as old states will not have type and we will not load it correctly. So it needs to be a separate if check that defaults to "".

Reply:
@NejcZdovc updated